flavours/arch/base.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# This script installs a base set of packages and configuration onto Arch Linux.
+# - `pacman -Syu`: Updates arch.
+# - `paru`: AUR helper that wraps pacman (https://github.com/morganamilo/paru).
+set -Exeuo pipefail
+
+# Upgrade system
+sudo pacman -Syu --noconfirm
+
+# Install paru
+sudo pacman -S --noconfirm --needed base-devel git cargo
+builddir=$(mktemp -d)
+sudo chown -R nobody:nobody "$builddir"
+cwd="$PWD"
+cd "$builddir"
+sudo -u nobody git clone https://aur.archlinux.org/paru.git
+cd "paru"
+export CARGO_HOME="$builddir/.cargo"
+sudo --preserve-env=CARGO_HOME -u nobody makepkg -s
+find "$builddir" -name "*.tar.zst" -print0 | xargs -n 1 -0 sudo -u nobody pacman --noconfirm -U
```

Comment:
@steakunderscore any ideas? :pray: I'm trying to install an AUR helper so I can install https://aur.archlinux.org/packages/chrome-remote-desktop or is there a way I can just install the `chrome-remote-desktop` AUR package directly?

Reply:
I'd be pretty tempted to skip the aur helpers and just install aur packages with `makepkg --install`

So roughly (without testing and mostly from memory) the steps would be:

```bash
sudo pacman -S --noconfirm --needed base-devel curl
for each package {
  curl -L -O https://aur.archlinux.org/cgit/aur.git/snapshot/package_name.tar.gz
  tar -xvf package_name.tar.gz
  cd package_name
  sudo  makepkg --install
  cd ../
  rm -rf package_name package_name.tar.gz
}
```

Generally aur helps are great for searching for packages, upgrading automatically etc. I doubt you need this.

One part I'm not super sure about, is it seems you can't run `sudo makepkg`. So maybe a hack would be to run `sudo -v && makepkg --install` or something like that. `makepkg` internally calls `sudo`.

docs: https://wiki.archlinux.org/title/Arch_User_Repository